### PR TITLE
14 Move file url function to api route

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on #1425 

Move all the computing URL logic from File class to ApiRoutes with the rest of the URL

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/data/api/ApiRoutes.kt [hunk 1]
```diff
@@ -97,6 +98,45 @@ object ApiRoutes {
...
+    //region File url
+    fun getDownloadFileUrl(file: File): String = with(file) {
+        val downloadUrl = if (isPublicShared()) {
+            downloadPublicShareFile(driveId, publicShareUuid, id)
+        } else {
+            downloadFile(file)
+        }
+
+        return if (isOnlyOfficePreview()) "$downloadUrl?as=pdf" else downloadUrl
+    }
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/data/api/ApiRoutes.kt:111]: What about doing extensions for these methods ?

For example:

```kotlin
fun File.getDownloadUrl(): String {
  val downloadUrl = if (isPublicShared()) {
      downloadPublicShareFile(driveId, publicShareUuid, id)
  } else {
      downloadFile(this)
  }

  return if (isOnlyOfficePreview()) "$downloadUrl?as=pdf" else downloadUrl
}
```

**Review by tevincent — CHANGES_REQUESTED** [review]

**Comment by tevincent** [comment]
Capture d’écran 2024-09-20 à 13 29 29

**Review by tevincent — APPROVED** [review]

**Review by tevincent — CHANGES_REQUESTED** [review]
131

File: app/src/main/java/com/infomaniak/drive/data/api/ApiRoutes.kt (hunk 1 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/data/api/ApiRoutes.kt:111]: If we put it as extension, we should put them in File.kt, that's a never ending question 😅

**Review by sirambd — APPROVED** [review]

**Review by tevincent — APPROVED** [review]

**Pull request merged**